jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1815,8 +1815,8 @@ public boolean equalsIgnoreCase(String string) {
 			char charAtO2Last = s2.charAtInternal(s1len - 1, s2Value);
 
 			if (charAtO1Last != charAtO2Last
-					&& toUpperCase(charAtO1Last) != toUpperCase(charAtO2Last)
-					&& ((charAtO1Last <= 255 && charAtO2Last <= 255) || Character.toLowerCase(charAtO1Last) != Character.toLowerCase(charAtO2Last))
+					&& (Character.toLowerCase(toUpperCase(charAtO1Last)) != Character.toLowerCase(toUpperCase(charAtO2Last)))
```

Comment:
As per https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#equalsIgnoreCase(java.lang.String), 
```
Two characters c1 and c2 are considered the same ignoring case if at least one of the following is true:

The two characters are the same (as compared by the == operator)
Calling Character.toLowerCase(Character.toUpperCase(char)) on each character produces the same result
```
`Character.toLowerCase(Character.toUpperCase(char))` can be used instead of `String. toUpperCase()`.